gap/base/methods.gd

```diff
@@ -0,0 +1,40 @@
+#############################################################################
+##
+##  This file is part of recog, a package for the GAP computer algebra system
+##  which provides a collection of methods for the constructive recognition
+##  of groups.
+##
+##  This files's authors include Sergio Siccha.
+##
+##  Copyright of recog belongs to its developers whose names are too numerous
+##  to list here. Please refer to the COPYRIGHT file for details.
+##
+##  SPDX-License-Identifier: GPL-3.0-or-later
+##
+##
+##  Declarations of recog methods
+##
+#############################################################################
+
+#! @Chapter methsel
+#! @Section whataremethods
+#! @Description
+#! TODO: how to get this into the chapter with the GAPDoc label "methsel"?
+#! TODO
+#! method     : the function itself
+#! rank       : an integer rank
+#! stamp      : a string describing the method uniquely
+#! comment    : an optional comment to describe the method for humans
+DeclareCategory("IsRecogMethod", IsObject);
+BindGlobal("RecogMethodFamily", NewFamily("RecogMethodFamily", IsRecogMethod));
+BindGlobal("RecogMethodType",
+           NewType(RecogMethodFamily,
+                   IsRecogMethod and IsAttributeStoringRep));
+
+# Functions and attributes for IsRecogMethod objects
+DeclareGlobalFunction("RecogMethod");
+DeclareGlobalFunction("CallRecogMethod");
+DeclareAttribute("FunctionOfRecogMethod", IsRecogMethod);
+DeclareAttribute("Stamp", IsRecogMethod);
+DeclareAttribute("Comment", IsRecogMethod);
```

Comment:
It's a bit odd that one has the `...OfRecogMethod` suffix but the others have not... I realize `Function` isn't a great name, though :-).

But anyway, why do we even need `FunctionOfRecogMethod `? Only `CallRecogMethod` needs it, right? So that could just access the value directly (you can store things into objects in other ways than using an attribute!).

Reply:
The purpose of the `FunctionOfRecogMethod` is to provide a documented access to the underlying function. I think people should have acces to that and I didn't want to document access the function via e.g. `m!.func`.
But I think instead of the attribute I'll store the function in `m!.func` and provide an access function `Unpack`.

Reply:
I've added a global function `UnpackRecogMethod`. I'm not married to the name though, so suggestions are always welcome. :)